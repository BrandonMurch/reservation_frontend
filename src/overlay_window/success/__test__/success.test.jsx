import React from 'react';
import { render } from '@testing-library/react';
import { unmountComponentAtNode } from 'react-dom';
import { create } from 'react-test-renderer';

import Success from '../index';

describe('<Success />', () => {
  let component;
  let container = null;

  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
    component = render(<Success />);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it('should match snapshot', () => {
    const tree = create(<Success />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should display success message', () => {
    const element = component.getByText(/Reservation was made successfully/i);
    expect(element).toBeInTheDocument();
  });
});
