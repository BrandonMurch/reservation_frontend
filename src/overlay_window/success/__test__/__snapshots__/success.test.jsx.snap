// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Success /> should match snapshot 1`] = `
<div
  className="container"
>
  <h1
    className="title"
  >
    Reservation was made successfully.
  </h1>
</div>
`;
